spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/notify/DingTalkNotifier.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2014-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.codecentric.boot.admin.server.notify;
+
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Base64;
+import org.springframework.context.expression.MapAccessor;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ParserContext;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestTemplate;
+import reactor.core.publisher.Mono;
+
+import de.codecentric.boot.admin.server.domain.entities.Instance;
+import de.codecentric.boot.admin.server.domain.entities.InstanceRepository;
+import de.codecentric.boot.admin.server.domain.events.InstanceEvent;
+
+/**
+ * Notifier submitting events to DingTalk.
+ *
+ * @author Mask
+ */
+public class DingTalkNotifier extends AbstractStatusChangeNotifier {
+
+	private static final String DEFAULT_MESSAGE = "#{instance.registration.name} #{instance.id} is #{event.statusInfo.status}";
+
+	private final SpelExpressionParser parser = new SpelExpressionParser();
+
+	private RestTemplate restTemplate;
+
+	/**
+	 * Webhook URI for the DingTalk API.
+	 */
+	private String webhookUrl;
+
+	/**
+	 * Secret for DingTalk.
+	 */
+	@Nullable
+	private String secret;
+
+	private Expression message;
+
+	public DingTalkNotifier(InstanceRepository repository, RestTemplate restTemplate) {
+		super(repository);
+		this.restTemplate = restTemplate;
+		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
```

Comment:
This is always using DEFAULT_MESSAGE, right? I cannot find any place where the text from property `spring.boot.admin.notify.dingtalk.message` is being used.

Reply:
This is a reference to other notifiers(eg. TelegramNotifier, SlackNotifier, PagerdutyNotifier, OpsGenieNotifier, LetsChatNotifier...).In oreder to be consistent with others notifiers,  So I did.

Reply:
For example :
In TelegramNotifier
> public TelegramNotifier(InstanceRepository repository, RestTemplate restTemplate) {
		super(repository);
		this.restTemplate = restTemplate;
		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
	}

In SlackNotifier

> public SlackNotifier(InstanceRepository repository, RestTemplate restTemplate) {
		super(repository);
		this.restTemplate = restTemplate;
		this.message = parser.parseExpression(DEFAULT_MESSAGE, ParserContext.TEMPLATE_EXPRESSION);
	}

Reply:
Hi, sorry for answering so late. These other notifiers have a setter for the description where it is parsed:

```
public void setMessage(String message) {
	this.message = parser.parseExpression(message, ParserContext.TEMPLATE_EXPRESSION);
}
```

Reply:
@mask616 if you add this piece we could merge your PR

Reply:
> @mask616 if you add this piece we could merge your PR

@erikpetzold I had add it just now.